common/statistics/src/gateways.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use nym_credentials_interface::TicketType;
+use nym_sphinx::DestinationAddressBytes;
+use time::OffsetDateTime;
+
+/// Channel for receiving incoming Stats events
+pub type GatewayStatsReceiver = tokio::sync::mpsc::UnboundedReceiver<GatewayStatsEvent>;
+
+/// Channel allowing for generic statistics events to be reported to a stats event aggregator.
+#[derive(Clone)]
+pub struct GatewayStatsReporter {
+    stats_tx: tokio::sync::mpsc::UnboundedSender<GatewayStatsEvent>,
+}
+
+impl GatewayStatsReporter {
+    /// Construct a new gateway statistics event reporter
+    pub fn new(stats_tx: tokio::sync::mpsc::UnboundedSender<GatewayStatsEvent>) -> Self {
+        Self { stats_tx }
+    }
+
+    /// Report a gateway statistivs event using the reporter
+    pub fn report(&self, event: GatewayStatsEvent) {
+        self.stats_tx.send(event).unwrap_or_else(|err| {
+            log::error!("Failed to report gateway stat event : {:?}", err);
+        });
+    }
+}
+
+/// Gateway Statistics events
+pub enum GatewayStatsEvent {
+    /// Events in the lifecycle of an established client tunnel
+    SessionStatsEvent(SessionEvent),
+}
+
+impl GatewayStatsEvent {
+    /// A new session between this gateway and the client remote has successfully opened
+    pub fn new_session_start(client: DestinationAddressBytes) -> GatewayStatsEvent {
+        GatewayStatsEvent::SessionStatsEvent(SessionEvent::SessionStart {
+            start_time: OffsetDateTime::now_utc(),
```

Comment:
in here you're not adjusting the event time to increase the anonymity set. shouldn't we do it everywhere?

Reply:
This one stays on gateway. When a session_end event is handled, we turn it into a duration, no need to replace timings here